Refine wording in Saddle Points (#2413)
exercism/problem-specifications: exercises/saddle-points/instructions.md

The data company provides the data as grids that show the heights of the trees.
The rows of the grid represent the east-west direction, and the columns represent the north-south direction.

> An acceptable tree will have the greatest height in its row, while having the least height in its column.

Author: I agree with @jordancurve that there is a risk here of the very same confusion that we are trying to avoid.

Reviewer: Isn't the largest or greatest height "tallest tree" and the least height being the "shortest tree", given the problem we are solving?

Author: The problem is that «the tallest tree» might not be unique. People have been frustrated by test cases with several tallest/shortest trees per row/column, because based on the instructions they expected there to be only one.

Reviewer: The tests should enforce the detail though, right?  So any ambiguity is clarified by expressed expectation, one would think.

Reviewer 2: The whole point of this PR is to clarify this point. It is okay fort the prose to omit a detail and for the test to enforce that detail. It is less okay for the prose to imply one thing and the test to enforce something that contradicts that thing.

Reviewer: The tests do seem to have covered multiple "shortest tree" at least in the single column test.  Or to express it in the way the test explains it "saddle points" meaning there can be more than one.
I wholly agree that the prose should not contradict, while omission is OK (not great, necessarily, but sometimes purposefully done).

I think that the detail that there may be more than one saddle point is expressed in the tests even by name.

We may see that here:

```json
    {
      "uuid": "ee99ccd2-a1f1-4283-ad39-f8c70f0cf594",
      "description": "Can identify that saddle points in a single column matrix are those with the minimum value",
      "property": "saddlePoints",
      "input": {
        "matrix": [
          [2],
          [1],
          [4],
          [1]
        ]
      },
      "expected": [
        {
          "row": 2,
          "column": 1
        },
        {
          "row": 4,
          "column": 1
        }
      ]
    },
    {
      "uuid": "63abf709-a84b-407f-a1b3-456638689713",
      "description": "Can identify that saddle points in a single row matrix are those with the maximum value",
      "property": "saddlePoints",
      "input": {
        "matrix": [
          [2, 5, 3, 5]
        ]
      },
      "expected": [
        {
          "row": 1,
          "column": 2
        },
        {
          "row": 1,
          "column": 4
        }
```

> An acceptable tree will have the greatest height in its row, while having the least height in its column.

Reviewer: ```suggestion
An acceptable tree will the tallestt in its row, while being the shortest  n its column.
```

Reviewer 3: @kotp "will the tallest" should that be "will be the tallest"?

Reviewer 3: @MatthijsBlom What about the above suggestion?

Reviewer 3: Couldn't we use the phrasing from the introduction?

Author: We could. I figured the difference between `introduction.md` and `instructions.md` might be deliberate, so I kept them different.

Reviewer 3: I'm fine with unifying them, especially when it turns out that the terminology is so hard to get right.

Author: If we use the same wording in both cases, we could also de-duplicate.

One potential advantage of using two different descriptions: these provide different perspectives on the same problem.

Author: I did not agree with applying this suggestion, but it was applied anyway.

The point of my change was to fix an inaccuracy. This now applied suggestion effectively reverts my change and _reintroduces_ the original inaccuracy. This inaccuracy has previously irritated people, hence this PR. See the forum thread, and the other threads here.
